var/spack/repos/builtin/packages/nek5000/package.py

```diff
@@ -0,0 +1,130 @@
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Nek5000(Package):
+    """A fast and scalable high-order solver for computational fluid
+       dynamics"""
+
+    homepage = "https://nek5000.mcs.anl.gov/"
+    url      = "https://github.com/Nek5000/Nek5000"
+
+    tags = ['cfd', 'flow', 'hpc', 'solver', 'navier-stokes',
+            'spectral-elements', 'fluid']
+
+    version('17.0.0-beta2', git='https://github.com/Nek5000/Nek5000.git',
+        commit='b95f46c59f017fff2fc19b66aa65a881085a7572')
+    version('develop',      git='https://github.com/Nek5000/Nek5000.git',
+        branch='master')
+
+    # MPI, Profiling and Visit variants
+    variant('mpi',       default=True, description='Build with MPI.')
+    variant('profiling', default=True, description='Build with profiling data.')
+    variant('visit',     default=False, description='Build with Visit.')
+
+    # Tools
+    variant('genbox',   default=True, description='Build genbox tool.')
+    variant('int_tp',   default=True, description='Build int_tp tool.')
+    variant('n2to3',    default=True, description='Build n2to3 tool.')
+    variant('postnek',  default=True, description='Build postnek tool.')
+    variant('reatore2', default=True, description='Build reatore2 tool.')
+    variant('genmap',   default=True, description='Build genmap tool.')
+    variant('nekmerge', default=True, description='Build nekmerge tool.')
+    variant('prenek',   default=True, description='Build prenek tool.')
+
+    depends_on('mpi', when="+mpi")
+    depends_on('visit', when="+vist")
+
+    @run_before('install')
+    def fortran_check(self):
+        if not self.compiler.fc:
+            msg = 'Cannot build Nek5000 without a Fortran compiler.'
+            raise RuntimeError(msg)
+
+    @run_before('install')
+    def c_check(self):
```

Comment:
Is this check needed? 
Does `spack` guarantee that there will always be a C compiler?

Reply:
You can set up a `compilers.yaml` without one, but I know of no configurations that would result in a Fortran compiler being configured without a C compiler.  I think yo8u can safely take this out.

Reply:
I will remove the C check.